src/main/scala/extracells/gui/GuiOreDictExport.java

```diff
@@ -38,37 +47,109 @@ public static void updateFilter(String _filter) {
 
 	public GuiOreDictExport(EntityPlayer player, PartOreDictExporter _part) {
 		super(new ContainerOreDictExport(player, _part));
+		this._containerOreDictExport = new ContainerOreDictExport(player, _part);
 		this.player = player;
 	}
 
 	@Override
 	protected void actionPerformed(GuiButton guibutton) {
 		new PacketOreDictExport(this.player, filter, Side.SERVER)
-				.sendPacketToServer();
+			.sendPacketToServer();
 	}
 
 	@Override
-	protected void drawGuiContainerBackgroundLayer(float f, int mouseX,
-			int mouseY) {
+	public void handleMouseInput() {
+		super.handleMouseInput();
+		deltaWheel = Mouse.getEventDWheel();
+		if (deltaWheel < 0) {
+			currentScroll++;
+		} else if (deltaWheel > 0) {
+			currentScroll--;
+		}
+	}
+	@Override
+	protected void drawGuiContainerBackgroundLayer(float f, int mouseX, int mouseY) {
 		drawDefaultBackground();
 		GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
 		Minecraft.getMinecraft().renderEngine.bindTexture(this.guiTexture);
 		drawTexturedModalRect(this.guiLeft, this.guiTop, 0, 0, this.xSize,
-				this.ySize);
+			this.ySize);
 		this.searchbar.drawTextBox();
 	}
 
+	public class RenderToolTip {
+		private ItemStack item;
+		private int x;
+		private int y;
+		private boolean render = false;
+
+		public RenderToolTip() {
+		}
+
+		public void setValue(ItemStack _item, int _x, int _y) {
+			this.item = _item;
+			this.x = _x;
+			this.y = _y;
+			this.render = true;
+		}
+
+		public void renderToolTip() {
+			if (render) {
+				GuiOreDictExport.this.renderToolTip(this.item, this.x, this.y);
+				this.render = false;
+			}
+
+		}
+
+	}
 	@Override
 	protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY) {
 		super.drawGuiContainerForegroundLayer(mouseX, mouseY);
-		this.fontRendererObj.drawString(
+		if (!Objects.equals(filter, "")) {
+			this.fontRendererObj.drawString(
 				StatCollector.translateToLocal(
-						"extracells.part.oredict.export.name").replace("ME ",
-						""), 8, 5, 0x000000);
-		this.fontRendererObj.drawString(
+					"extracells.part.oredict.export.name").replace("ME ",
+					""), 8, 5, 0x000000);
+			this.fontRendererObj.drawString(
 				StatCollector.translateToLocal("container.inventory"), 8,
 				this.ySize - 94, 0x000000);
+			this._containerOreDictExport.part.setFilter(filter, false);
+			Iterator<AEItemStack> items = this._containerOreDictExport.part.getOres().iterator();
+			int size = this._containerOreDictExport.part.getOres().size();
 
+			if (this.currentScroll < 0)
+				this.currentScroll = 0;
+			int maxPage = size / (4 * 10);
+			if (this.currentScroll > maxPage) {
+				this.currentScroll = maxPage;
+			}
+			for (int i = 0; i < 10 * 4 * currentScroll; i++) {
```

Comment:
Have you tested how slow this is if the filter is something like `*` or `wood*` where there are many items in the full pack? Maybe a hard cap on the total number of items shown is needed